### PR TITLE
FIX: App crash when local node is down #7027

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: blue_modules/BlueElectrum.ts [hunk 1]
```diff
@@ -194,12 +194,25 @@ export async function connectMain(): Promise<void> {
...
   }
+
   let usingPeer = getNextPeer();
   const savedPeer = await getSavedPeer();
   if (savedPeer && savedPeer.host && (savedPeer.tcp || savedPeer.ssl)) {
     usingPeer = savedPeer;
   }
 
+  // Check if the peer is online before attempting the connection
+  const isOnline = await testConnection(
```
Comment (Overtorment) [inline blue_modules/BlueElectrum.ts:205]: i suspect the issue is something else. `testConnection` uses the same mechanism to connect, so it should crash too.

plus its not even supposed to crash, connection is inside try-catch, and we handle bad connection pretty well, by having `onError` handler, offering dialog with retries, stop, or change electrum server.

File: blue_modules/BlueElectrum.ts [hunk 2]
```diff
@@ -224,53 +236,37 @@ export async function connectMain(): Promise<void> {
     mainClient.onError = function (e: { message: string }) {
       console.log('electrum mainClient.onError():', e.message);
       if (mainConnected) {
-        // most likely got a timeout from electrum ping. lets reconnect
```
Comment (Overtorment) [inline blue_modules/BlueElectrum.ts:227]: useful coment gone

File: blue_modules/BlueElectrum.ts [hunk 3]
```diff
@@ -213,7 +226,6 @@ export async function connectMain(): Promise<void> {
       await DefaultPreference.set(ELECTRUM_SSL_PORT, usingPeer.ssl ?? '');
     }
   } catch (e) {
-    // Must be running on Android
```
Comment (Overtorment) [inline blue_modules/BlueElectrum.ts:216]: useful coment gone

File: blue_modules/BlueElectrum.ts [hunk 4]
```diff
@@ -224,53 +236,37 @@ export async function connectMain(): Promise<void> {
...
-          case 'electrs-esplora':
-            // its a different one, and it does NOT support batching
-            // nop
-            break;
-          case 'Fulcrum':
-            if (semVerToInt(electrumVersion) >= semVerToInt('1.9.0')) {
-              disableBatching = false;
-            }
-            break;
+        if (electrumImplementation === 'electrs' && semVerToInt(electrumVersion) >= semVerToInt('0.9.0')) {
```
Comment (Overtorment) [inline blue_modules/BlueElectrum.ts:256]: switch-case is way more readable. bad.

File: blue_modules/BlueElectrum.ts [hunk 5]
```diff
@@ -279,7 +275,7 @@ export async function connectMain(): Promise<void> {
 
   if (!mainConnected) {
     console.log('retry');
-    connectionAttempt = connectionAttempt + 1;
+    connectionAttempt += 1;
```
Comment (Overtorment) [inline blue_modules/BlueElectrum.ts:278]: change for a sake of change. did chatgpt refactor this file?

**Review by Overtorment — CHANGES_REQUESTED** [review]
shitpost quality pr

**Comment by Overtorment** [comment]
closing as root cause of crashes is unclear

**Pull request closed without merging**